src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.ts

```diff
@@ -1 +1,132 @@
+import { AlterationTypeConstants } from '../../../pages/resultsView/ResultsViewPageStore';
+import client from 'shared/api/cbioportalClientInstance';
+import {
+    GenericAssayMetaFilter,
+    GenericAssayDataFilter,
+    GenericAssayData,
+    GenericAssayMeta,
+} from 'shared/api/generated/CBioPortalAPI';
+import { MolecularProfile } from 'shared/api/generated/CBioPortalAPI';
+import _ from 'lodash';
+import { IDataQueryFilter } from '../StoreUtils';
+import { GenericAssayDataMultipleStudyFilter } from 'shared/api/generated/CBioPortalAPI';
+
 export const NOT_APPLICABLE_VALUE = 'NA';
+
+export async function fetchGenericAssayMetaByMolecularProfileIdsGroupByGenericAssayType(
+    molecularProfiles: MolecularProfile[]
+) {
+    const genericAssayProfiles = molecularProfiles.filter(profile => {
+        return (
+            profile.molecularAlterationType ===
+            AlterationTypeConstants.GENERIC_ASSAY
+        );
+    });
+
+    const genericAssayProfilesGroupByGenericAssayType = _.groupBy(
+        genericAssayProfiles,
+        'genericAssayType'
+    );
+    const genericAssayTypes = _.keys(
+        genericAssayProfilesGroupByGenericAssayType
+    );
+
+    const GenericAssayMetaGroupByGenericAssayType: {
+        [genericAssayType: string]: GenericAssayMeta[];
+    } = {};
+
+    await Promise.all(
```

Comment:
this one is kinda weird.  i'm not totally sure why it works because the map function doesn't return a promise as far as I can see.  Promise.all needs an array of promises i think.  i'm not sure why tyepscript allows this.   anyways, you should assign the result of await Promise.all and then use resulting array to populate GenericAssayMetaGroupByGenericAssayType.

Reply:
I just checked in the console and passing nonpromises to Promise.all is acceptable.

However that's not what's happening here because `async` causes a function to return a promise

Reply:
After discussion I agree with @alisman and think something like my suggestion below would be most clear here too